File: google/main/src/core/structures/Token.java
package main.src.core.structures;

public class Token extends Entity {

    public Token(String id) {
        super(id);
        this.id = normalize(id);
    }

    private static String normalize(String str) {
        String normalized = str;
        normalized = str.toLowerCase();
        return normalized;
    }

    @Override
    public void setId(String newId){
        super.setId(normalize(newId));
    }

    @Override
    public String toString() {
        return "Token(" + this.id + ")";
    }
}

File: BagherMusic/server/Models/Token.cs
// Standard Library
using System;

namespace BagherMusic.Models
{
	public class Token : IEntity<string>
	{
		private string id;

		public Token(string id)
		{
			Id = id;
		}

		public new string Id
		{
			get
			{
				return id;
			}
			set
			{
				id = Normalize(value);
			}
		}

		public override string ToString()
		{
			return $"Token({Id})";
		}

		private static string Normalize(string key)
		{
			string normalized = key;
			normalized = normalized.ToLower();
			return normalized;
		}
	}
}
